polars/polars-time/src/windows/groupby.rs

```diff
@@ -525,49 +525,47 @@ pub fn groupby_values(
 
     // we have a (partial) lookbehind window
     if offset.negative {
-        if offset.duration_ns() >= period.duration_ns() {
```

Comment:
here, previously there were two paths:

- `offset` >= `period`, `offset` < `period * 2`: groupby_values_iter_full_lookbehind
- `offset` >= `period`, `offset` >= `period * 2`: groupby_values_iter_window_behind_t
- `offset` < `period`: groupby_values_iter_partial_lookbehind

Just changing it to
- `offset` >= `period`: groupby_values_iter_window_behind_t
- `offset` < `period`: groupby_values_iter_partial_lookbehind

works, but the comments suggest `groupby_values_iter_window_behind_t` is less performant than `groupby_values_iter_full_lookbehind`. It is still possible to use the latter, but only in some cases - I need to look into this carefully

Reply:
I belive `groupby_values_iter_full_lookbehind` assumes that `t` is completely behind the window. So there are more cases where we have that besides `period == offset`. 

I will have to dive into it which cases it were again. Do you have on top of mind which predicate would inlcude all cases where `t` is full lookbehind?

This is beneficial as in that case we can parallelize over `t` and then look from that point backwards in the slice to find the window.

Reply:
Right, let's first make it correct. We can try to find fast paths later if needed. :+1:

Thanks!